kernel/src/drivers/ps2/mod.rs

```diff
@@ -90,167 +93,159 @@ impl Ps2Controller {
             println!("ps2c: detected no available devices");
         }
 
-        flush_output();
+        flush_output()?;
 
         self.initialized = true;
+
+        Ok(())
     }
 
     /// Initializes the config for this controller
-    fn initialize_config(&mut self) {
+    fn initialize_config(&mut self) -> Result<(), Ps2Error> {
         // Read the config from the controller
-        self.read_config();
+        self.read_config()?;
 
         // Set all required config flags
-        self.config.set(ControllerConfigBit::PortInterrupt1, false);
-        self.config.set(ControllerConfigBit::PortInterrupt2, false);
-        self.config.set(ControllerConfigBit::PortTranslation1, false);
+        self.config.set(Ps2ConfigFlags::PORT_INTERRUPT_1, false);
+        self.config.set(Ps2ConfigFlags::PORT_INTERRUPT_2, false);
+        self.config.set(Ps2ConfigFlags::PORT_TRANSLATION_1, false);
 
         // Write the updated config back to the controller
-        self.write_config();
+        self.write_config()?;
 
         println!("ps2c: initialized config");
+
+        Ok(())
     }
 
     /// Tests this controller
-    fn test_controller(&mut self) -> bool {
-        command_ret(ControllerReturnCommand::TestController) == Some(0x55)
+    fn test_controller(&mut self) -> Result<bool, Ps2Error> {
+        Ok(command_ret(ControllerReturnCommand::TestController)? == 0x55)
     }
 
     /// Writes the current config to the PS2 controller
-    pub fn write_config(&mut self) {
-        command_data(ControllerCommand::WriteConfig, self.config.data);
+    pub fn write_config(&mut self) -> Result<(), Ps2Error> {
```

Comment:
See comment earlier about this

Reply:
I don't understand how that could apply here. Do you mean the config should rather be passed to `write_config`?